sdk/core/azure-core/test/ut/bodystream_test.cpp

```diff
@@ -145,3 +145,97 @@ TEST(FileBodyStream, Read)
   EXPECT_EQ(readSize, FileSize);
   EXPECT_EQ(buffer[FileSize], 0);
 }
+
+TEST(ProgressBodyStream, Init)
+{
+  int64_t bytesTransferred = -1;
+  std::string testDataPath(AZURE_TEST_DATA_PATH);
+  testDataPath.append("/fileData");
+
+  Azure::Core::IO::FileBodyStream stream(testDataPath);
+
+  ProgressBodyStream progress(stream, [&bytesTransferred](int64_t bt) { bytesTransferred = bt; });
+
+  EXPECT_EQ(bytesTransferred, 0);
+  EXPECT_EQ(progress.Length(), stream.Length());
+}
+
+TEST(ProgressBodyStream, ReadChunk)
+{
+  int64_t bytesTransferred = -1;
+  std::string testDataPath(AZURE_TEST_DATA_PATH);
+  testDataPath.append("/fileData");
+
+  Azure::Core::IO::FileBodyStream stream(testDataPath);
+
+  ProgressBodyStream progress(stream, [&bytesTransferred](int64_t bt) { bytesTransferred = bt; });
+
+  std::vector<uint8_t> buffer(10);
+
+  size_t readSize = progress.ReadToCount(buffer.data(), 10);
```

Comment:
nit: maybe?

```suggestion
  size_t readSize = progress.ReadToCount(buffer.data(), buffer.size());
```

Reply:
in that test i just read a chunk , i increased the size of the stream to make it more clear